lib/matplotlib/artist.py

```diff
@@ -461,28 +461,18 @@ def get_children(self):
         r"""Return a list of the child `.Artist`\s of this `.Artist`."""
         return []
 
-    def _default_contains(self, mouseevent, figure=None):
+    def _not_same_canvas(self, event):
```

Comment:
Is there a particular reason to include the negation in the function? Typically one would do `if not self._same_canvas(event)`. That prevents possible double negations in case one wants to check “same canvas” in the future.

Reply:
Yes, because if we can only know if the event for sure happened in a different canvas, but cannot know for sure that it happened in the same canvas (because either artist.figure can be done -- this occurs for some sub-artists -- or event.canvas can be none -- this happens for synthetic events).
So not_same_canvas = True: definitely not the same; not_same_canvas = False: perhaps, not sure.
This seems better than same_canvas = False: definitely not the same; same_canvas = True: perhaps the same.

Reply:
Thanks for the explanation. In that case I propose

1. Naming: `def _different_canvas(self, event)` and
2. Explicitly mention in the docstring that False means "not sure" so that `not _different_canvas()` does not imply "same".

Alternatively one could `def _certainly_different_canvas()`, which would convey both meanings correctly "certainly different" / "not certainly different". But I'm not sure myself whether that's too long or too clever.

Reply:
I like Tim's suggestion of `_different_canvas()`, it is easier to comprehend when looking quickly at the code.

Reply:
Sure, done.